libmamba/include/mamba/download/mirror.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_DOWNLOAD_MIRROR_HPP
+#define MAMBA_DOWNLOAD_MIRROR_HPP
+
+#include <functional>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <fmt/core.h>
+
+#include "mamba/download/request.hpp"
+
+namespace mamba::download
+{
+    class MirrorID
```

Comment:
Should this be a plain struct?

Reply:
No, because the internal representation should never been accessed by client code.

Reply:
Yes, we need this to not change interface in the future but stilll be optimizable.